### ISSUE TITLE
cannot use python 3.9 with tensorflow==1.15

### ISSUE THREAD

**Issue by mschrimpf** [brain-score/model-tools#47]
We're currently still on TensorFlow 1: https://github.com/brain-score/model-tools/blob/291970092ce439569b9e063324254c1d8c4702d3/setup.py#L18
Python 3.9 so far does not seem to work with 1.15, we should upgrade to TensorFlow 2 to avoid specifying versions directly

**Comment by raj-magesh** [comment]
Is there a timeline for this upgrade? I need some TensorFlow 2 functionality that I can't easily access in the same environment because model-tools forces TensorFlow 1.15.

**Comment by mschrimpf** [comment]
So far, we did not have this planned until mid 2022, but if there's a necessity we will prioritize it sooner.

Can you please describe your use case? Is there a way to get away with tf1 and e.g. re-defining the functionality you're using?

**Comment by martinosorb** [comment]
For what is tensorflow used in the library? I had the same problem and I simply told pip to ignore the tensorflow dependency while installing model-tools. The tests seem to work. Of course it may be needed for other things which I'm not currently using...

**Comment by mschrimpf** [comment]
It's actually only used in the [OST](https://github.com/brain-score/brain-score/blob/9800ae7dd3ab88bbe68e0be74a569193076947fc/brainscore/metrics/ost.py) metric which isn't included in the competition (as well as the [mask_regression](https://github.com/brain-score/brain-score/blob/0e34c8389d912ae7be9c4012abdb52e4a62c51ce/brainscore/metrics/mask_regression.py) metric which currently isn't used at all for the standard benchmarks). 
Did you already submit? It might be that the model fails on the Kar2019-ost benchmark but should work on the others.

**Comment by raj-magesh** [comment]
Sorry for the late reply! I needed a TF2 utility to convert pretrained TF models to PyTorch. I decided to just create a separate environment for that step of my workflow, so this isn't urgent. Closing the issue; thanks for the responses!

**Comment by YudiXie** [comment]
I got an issue related to this when installing this repo on a MacBook with Apple Silicon. It seems that the issue originated from the fact that this package only uses tensorflow==1.15, which is not compatible with Python 3.8 or higher.

However, the newer Anaconda distribution for the M1 MacOS system only has Python 3.8 and higher, making it hard to run this repo on the laptop. So to install this package, I may have to reinstall Anaconda or switch to another machine that supports Python 3.7.

It would be really great if this repo could support newer versions of TensorFlow and Python.

**Comment by nikparth** [comment]
to add to this, there are a lot of models these days that only have weights in TF2. Unless there is an easy way to convert these to previous TF versions or pytorch (which I haven't seen a general way to do), this limits the models that can be evaluated.

**Comment by IanCarrasco** [comment]
Any updates on this dependency? I've been interested in getting this up and running and have gone through various python versions/venvs with no luck. Thank you!

**Comment by IanCarrasco** [comment]
It appears that this PR updated the two files in question to use TF2 (https://github.com/brain-score/brain-score/pull/196/commits/04b01f52414553c18961d06e2acec93d78f76968), but it may just be an old dependency in model-tools? If those were the last two TF1 references. 

Potentially Old Dependency in Question:
https://github.com/brain-score/model-tools/blob/master/setup.py#L18C5-L18C24

**Comment by mschrimpf** [comment]
hi all, as an update to this: we have just released a major refactor which combines `model-tools` and `models` into `brain-score`: https://github.com/brain-score/vision

This should make it a lot easier to fix the tensorflow issues and potentially upgrade python.
